autoscheduler/user_sessions/views.py

```diff
@@ -97,6 +98,49 @@ def save_availabilities(request):
     """ Saves availabilities for the given user in the session """
     return _set_state_in_session(request, 'availabilities')
 
+@api_view(['GET'])
+def get_saved_schedules(request):
+    """ Returns the saved schedules from the session for the requested term """
+    term = request.query_params.get('term')
+
+    if not term:
+        return Response(status=400)
+
+    with retrieve_data_session(request) as data_session:
+        schedules = data_session.get(term, {}).get('schedules')
+
+        if not schedules:
+            return Response([])
+
+        # Gather all of the section IDs
+        section_ids = set(sec for schedule in schedules for sec in schedule['sections'])
+
+        models = Section.objects.filter(
+            id__in=section_ids
+        ).select_related('instructor').prefetch_related('meetings')
+
+        sections_dict = {section.id: section for section in models.iterator()}
+
+        def serialize_schedule(sections):
+            sections = set(sections_dict[section] for section in sections)
+
+            return SectionSerializer(
+                sections, many=True, context={'skip_grades': True}
+            ).data
+
+        ret = [{
+            'name': schedule['name'],
+            'sections': serialize_schedule(schedule['sections']),
+        } for schedule in schedules]
```

Comment:
Is this not exactly the same as `_serialize_schedules` in the scheduler? If not, why does it have to be different and if so, why is it repeated here instead of imported

Reply:
Their input and output is slightly different which is why I didn't do it initially, but I'll change it to use it + apply all of the optimizations to it (removing the `models.iterator()`, prefetch, etc)